Documentation/upgrading.md

```diff
@@ -1,173 +1,149 @@
+Let's upgrade a Kubernetes v1.6.6 cluster to v1.6.7 as an example.
 
-Let's upgrade a self-hosted Kubernetes v1.4.1 cluster to v1.4.3 as an example.
+## Status
 
-## Inspect
-
-Show the control plane daemonsets and deployments which will need to be updated.
-
-    $ kubectl get daemonsets -n=kube-system
-    NAME             DESIRED   CURRENT   NODE-SELECTOR                    AGE
-    kube-apiserver   1         1         node-role.kubernetes.io/master   5m
-    kube-proxy       3         3         <none>                           5m
-    kubelet          3         3         <none>                           5m
-
-    $ kubectl get deployments -n=kube-system
-    NAME                      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
-    kube-controller-manager   1         1         1            1           5m
-    kube-dns-v20              1         1         1            1           5m
-    kube-scheduler            1         1         1            1           5m
-
-Check the current Kubernetes version.
-
-    $ kubectl version
-    Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.0", GitCommit:"a16c0a7f71a6f93c7e0f222d961f4675cd97a46b", GitTreeState:"clean", BuildDate:"2016-09-26T18:16:57Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
-    Server Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.1+coreos.0", GitCommit:"b7a02f46b972c5211e5c04fdb1d5b86ac16c00eb", GitTreeState:"clean", BuildDate:"2016-10-11T20:13:55Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
-
-In this case, Kubernetes is `v1.4.1+coreos.0` and our goal is to upgrade to `v1.4.3+coreos.0`. First, update the control plane pods. Then the kubelets and proxies on all nodes.
-
-## Control Plane
-
-### kube-apiserver
+bootkube does not provide in-place patch diffs or migrations between releases. In-place upgrades are inherently more dangerous than a blue/green cluster deployments. Evaluate whether an in-place upgrade is appropriate for your Kubernetes cluster and risk tolerance before beginning. Failover important workloads to another cluster in advance.
 
-Edit the kube-apiserver daemonset. Change the container image name to `quay.io/coreos/hyperkube:v1.4.3_coreos.0`.
+## Prepare
 
-    $ kubectl edit daemonset kube-apiserver -n=kube-system
+Find the diff between bootkube assets between the existing cluster version and the desired cluster version. This depends on the tool used to generate assets:
 
-Since daemonsets don't yet support rolling, manually delete each apiserver one by one and wait for each to be re-scheduled.
+* Manual - Check Github [compare](https://github.com/kubernetes-incubator/bootkube/compare/v0.6.0...v0.6.1) between existing and diesred versions and infer the appropriate cluster changes
```

Comment:
Can we use the diff in the example (v1.6.6, v1.6.7)

Reply:
So that would be v0.5.0...v0.5.1 if we wanna correspond to the Kubernetes v1.6.6 to v1.6.7 updates -  its comparing the assets produced at different bootkube versions corresponding to Kubernetes versions.

Reply:
oh woops. But yeah - probably can't hurt to compare the actual bootkube versions to the example k8s versions.